website/docs/tutorials/metatype-basics/index.mdx

```diff
@@ -1,118 +1,1080 @@
-:::
+Looking through the app we can see tha the api should allow:
+ - Unauthenticated users to submit new "ideas" or vote on any of those already listed.
+ - Specify or vote on the importance of an "idea" from "medium" to "critical" or even submit text with more description.
+ - Admins will be able to move ideas across buckets like "Backlog", "Planned", "In Progress".
 
-## Zooming on the types
+        {/* TODO: consider emailing people interested in an idea when things change */}
 
-There is no "object" or "primitive" type, only 4 main categories of types:
+## Setup
 
-- value types: `t.integer()`, `t.string()`, `t.uuid()`, etc.
-- quantifier types: `t.optional(·)`, `t.list(·)`, etc.
-- consolidator types: `t.struct(·, ·)`, `t.union(·, ·)`, etc.
-- function types: `t.func(· → ·)`, `t.policy(· → ·)`, etc.
+Before anything, we'll first need to install the tooling required for development and create a new project. 
 
-You can combine them with each other to describe almost any data type you may need. The typegate enforces the data validation when data flows through it. Some syntactic sugar is available to make the type definition shorter:
+### Install Meta CLI
 
-    <CodeBlock language="python">{require("./types.py").content}</CodeBlock>
+The meta-cli tool manages and helps you develop your metatype based projects. 
+This include allowing you to run them locally for development, push them to production in the cloud, managing your database migrations and more.
 
-## The typegraph package
+        <InstallMetaCli />
 
-The typegraph package is a Python package that allows to describe a full typegraph. It's a thin wrapper around the type system, and provides a few helpers to make the typegraph definition easier. It builds on the type system to provide some more building blocks:
+### Create a new project
 
-1. _what_ data types exists → value, quantifier, consolidator **types**
-2. _how_ these data get transformed → function types and **materializers** that specify the transformation
-3. _where_ these data and transformations run → **runtimes** that describe materializers operate
-4. _who_ can access them → a special case of function types named **policies** that control accesses
+Metatype projects are composed of modular bundles of types, logic and policies called typegraphs.
+We author typegraphs using modern programming languages & environments. 
+Python and ECMAScript/Typescript are currently available for use.
+The meta-cli allows us to create a new project based on pre-existing templates.
 
-This tutorial will cover these abstractions concept by concept and show how to use them by example.
 
-:::info Code sample are interactive
-You can interact with most of the typegraph in the documentation. Press command/ctrl + enter to submit the selected query or use the "play" button. Some parts might be voluntarily hidden and full source can be found by clicking on the link.
-:::
+        <SDKTabs>
+                      <TabItem value="typescript">
```

Comment:
The indentation seems of here, and does not align to the ending tag.

Reply:
Yeah, that's some weirdness around the mdx formatter in vscode. I can't seem to find it but docusaurus MDX needs some special handling around tabs (notice the extra comment block before the content starts) and mabye the weird padding's it's intentional. I would fix it but I'd done so multiple times only to be foiled by the formatter so I'm just letting it be lol